signer/cosigner_nonce_cache.go

```diff
@@ -62,48 +60,28 @@ func (lc *lastCount) Get() int {
 }
 
 type NonceCachePruner interface {
-	PruneNonces()
+	PruneNonces() int
 }
 
 type NonceCache struct {
-	cache map[uuid.UUID]*CachedNonce
+	cache []*CachedNonce
 	mu    sync.RWMutex
 }
 
-func NewNonceCache() NonceCache {
-	return NonceCache{
-		cache: make(map[uuid.UUID]*CachedNonce, cachePreSize),
-	}
-}
-
 func (nc *NonceCache) Size() int {
 	nc.mu.RLock()
 	defer nc.mu.RUnlock()
 	return len(nc.cache)
 }
 
-func (nc *NonceCache) Get(uuid uuid.UUID) (*CachedNonce, bool) {
-	nc.mu.RLock()
-	defer nc.mu.RUnlock()
-	cn, ok := nc.cache[uuid]
-	return cn, ok
-}
-
-func (nc *NonceCache) Set(uuid uuid.UUID, cn *CachedNonce) {
+func (nc *NonceCache) Add(cn *CachedNonce) {
 	nc.mu.Lock()
 	defer nc.mu.Unlock()
-	nc.cache[uuid] = cn
+	nc.cache = append(nc.cache, cn)
 }
 
-func (nc *NonceCache) GetSortedByExpiration() []*CachedNonce {
-	cns := make([]*CachedNonce, 0, len(nc.cache))
-	for _, cn := range nc.cache {
-		cns = append(cns, cn)
-	}
-	sort.Slice(cns, func(i, j int) bool {
-		return cns[i].Expiration.Before(cns[j].Expiration)
-	})
-	return cns
+func (nc *NonceCache) Delete(index int) {
+	nc.cache = append(nc.cache[:index], nc.cache[index+1:]...)
```

Comment:
Does this succeed for both the first and last index in the slice?

Reply:
yes, added unit test